docs/csharp/tutorials/working-with-linq.md

```diff
@@ -270,62 +313,22 @@ public static void Main(string[] args)
-                    from r in Ranks().LogQuery("Value Generation")
-                    select new PlayingCard(s, r))
-                    .LogQuery("Starting Deck")
-                    .ToArray();
-```
+Please note that this example is **designed** to highlight the use cases where lazy evaluation can cause performance difficulties. While it's important to see where lazy evaluation can impact code performance, it's equally important to understand that not all queries should run eagerly. The performance hit we incur without using `ToArray()` is because each new arrangement of the deck of cards is built from the previous arrangement. Using lazy evaluation means each new deck configuration is built from the original deck, even executing the code that built the `startingDeck`. That causes a large amount of extra work. 
 
-Compile and run again. The output is a little cleaner, and the code is a bit more clear and can be extended more easily.
+In practice, some algorithms run well using eager evaluation, and others run well using lazy evaluation. For daily usage, lazy evaluation is usually a better choice when the data source is a separate process, like a database engine. For databases, lazy evaluation allows more complex queries to execute only one round trip to the database process and back to the rest of your code. LINQ is flexible whether you choose to utilize lazy or eager evaluation, so measure your processes and pick whichever kind of evaluation gives you the best performance.
 
 ## Conclusion
 
-This sample showed you some of the methods used in LINQ, how to create your own methods that will be easily used with LINQ enabled code. It also showed you the differences between lazy and eager evaluation, and the effect that decision can have on performance.
+In this project, we covered:
+* using LINQ queries to aggregate data into a meaningful sequence
+* writing Extension methods to add our own custom functionality to LINQ queries
+* locating areas in our code where our LINQ queries might run into performance issues like degraded speed
+* lazy and eager evaluation in regards to LINQ queries and the implications they might have on query performance
 
-You learned a bit about one magician's technique. Magicians use the faro shuffle because they can control where every card moves in the deck. In some tricks, the magician has an audience member place a card on top of the deck, and shuffles a few times, knowing where that card goes. Other illusions require the deck set a certain way. A magician will set the deck prior to performing the trick. Then she will shuffle the deck 5 times using an outer shuffle. On stage, she can show what looks like a random deck, shuffle it 3 more times, and have the deck set exactly how she wants.
+Aside from LINQ, we learned a bit about a technique magicians use for card tricks. Magicians use the Faro shuffle because they can control where every card moves in the deck. Now that you know, don't spoil it for everyone else!
```

Comment:
Docs contain many more examples of LINQ usage. I think they should be referenced from this tutorial in the **See also** section. Articles to reference might be:
https://docs.microsoft.com/en-us/dotnet/csharp/linq/
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/

Reply:
I have a few pages in mind, but how would I link those two pages? I don't think I can reference folders using traditional markdown syntax, i.e.:

`[Language Integrated Query (LINQ)](../the-linq-page)`

Where "the-linq-page" = https://docs.microsoft.com/en-us/dotnet/csharp/linq/
Would this just be a use case for hyperlinks?

Reply:
@jmarkman cool, thanks!

> I'd almost worry that docs pages would begin to circularly reference each other.

I don't think it's a big issue. Sometimes, I need to read the same page twice, with a break between reading sessions, to get full understanding of the topic. Circular referencing fits that workflow 😄